samples/python/calibrate.py

```diff
@@ -160,14 +164,42 @@ def processImage(fn):
         from multiprocessing.dummy import Pool as ThreadPool
         pool = ThreadPool(threads_num)
         chessboards = pool.map(processImage, img_names)
-
+        
+    def matchImagePointsforcharuco(board, charuco_corners, charuco_ids):
+        obj_pts = []
+        img_pts = []
+        for i in range(0, len(charuco_ids)):
+            index = charuco_ids[i]
+            obj_pts.append(board.getChessboardCorners()[index])
+            img_pts.append(charuco_corners[i])
+
+        obj_pts = np.array(obj_pts)
+        img_pts = np.array(img_pts)
+
+        return obj_pts, img_pts
+    
     chessboards = [x for x in chessboards if x is not None]
+
+    # not using pattern_points
     for (corners, pattern_points) in chessboards:
-        img_points.append(corners)
-        obj_points.append(pattern_points)
+        obj_points_tmp, img_points_tmp = matchImagePointsforcharuco(board, corners, _charucoIds)
```

Comment:
There is matchImagePoints method for it: https://docs.opencv.org/4.x/d4/db2/classcv_1_1aruco_1_1Board.html#a305c3c47c1f87562ed06de5c7e703e4f